src/main/java/org/elasticsearch/example/realm/CustomAuthenticationFailureHandler.java

```diff
@@ -98,4 +96,36 @@ public ElasticsearchSecurityException authenticationRequired(String action) {
         se.addHeader("WWW-Authenticate", "custom-challenge");
         return se;
     }
+
+    /**
+     * Allows you to allow the internal realms (or even other custom realms)
```

Comment:
Let's reword this a bit:

```
This method handles the logic for adding a custom header when there is a missing token, which includes the ability to preserve existing authenticate challenges (WWW-Authenticate headers) or to overwrite them
```

There is only one failure handler, so there will be no other custom realms that add these headers